pkg/destroy/powervs/cloudobjectstorage.go

```diff
@@ -0,0 +1,226 @@
+package powervs
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/IBM/go-sdk-core/v5/core"
+	"github.com/IBM/platform-services-go-sdk/resourcecontrollerv2"
+	"github.com/pkg/errors"
+)
+
+const cosTypeName = "cos instance"
+const cosResourceID = "dff97f5c-bc5e-4455-b470-411c3edbe49c"
+
+// listCOSInstances lists COS service instances.
+func (o *ClusterUninstaller) listCOSInstances() (cloudResources, error) {
+	o.Logger.Debugf("Listing COS instances")
+
+	ctx, _ := o.contextWithTimeout()
+
+	options := o.controllerSvc.NewListResourceInstancesOptions()
+	options.SetResourceID(cosResourceID)
+	options.SetType("service_instance")
```

Comment:
if you know the resource name then set the Name in the options to optimise the result

Reply:
you can also set the `ResourceGroupID` to reduce the scope to resource group(with assumption that we know the resource group we are using for the cluster creation)